src/ReverseProxy.ServiceFabric/ServiceDiscovery/Util/LabelsParser.cs

```diff
@@ -140,53 +169,60 @@ internal static List<ProxyRoute> BuildRoutes(Uri serviceName, Dictionary<string,
+                        var transformName = keyRemainder.Slice(0, transformNameLength).ToString();
                         if (!_allowedTransformNamesRegex.IsMatch(transformName))
                         {
                             throw new ConfigException($"Invalid transform index '{transformName}', should be transform index wrapped in square brackets.");
                         }
-                        if (!transforms.ContainsKey(transformName)) 
+                        if (!transforms.ContainsKey(transformName))
                         {
                             transforms.Add(transformName, new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase));
                         }
-                        var propertyName = kvp.Key.Substring($"{thisRoutePrefix}.Transforms.{transformName}.".Length);
-                        if (!transforms[transformName].ContainsKey(propertyName)) 
+                        var propertyName = keyRemainder.Slice(transformNameLength + 1).ToString();
+                        if (!transforms[transformName].ContainsKey(propertyName))
                         {
                             transforms[transformName].Add(propertyName, kvp.Value);
-                        } 
-                        else 
+                        }
+                        else
                         {
                             throw new ConfigException($"A duplicate transformation property '{transformName}.{propertyName}' was found.");
                         }
                     }
+                    else if (ContainsKey("Hosts", routeLabelKey, out _))
+                    {
+                        hosts = kvp.Value;
+                    }
+                    else if (ContainsKey("Path", routeLabelKey, out _))
+                    {
+                        path = kvp.Value;
+                    }
+                    else if (ContainsKey("Order", routeLabelKey, out _))
+                    {
+                        order = ConvertLabelValue<int?>(kvp.Key, kvp.Value);
+                    }
                 }
-                
-
-                labels.TryGetValue($"{thisRoutePrefix}.Hosts", out var hosts);
-                labels.TryGetValue($"{thisRoutePrefix}.Path", out var path);
 
                 var route = new ProxyRoute
                 {
-                    RouteId = $"{Uri.EscapeDataString(backendId)}:{Uri.EscapeDataString(routeName)}",
+                    RouteId = $"{Uri.EscapeDataString(backendId)}:{Uri.EscapeDataString(routeNamePair.Value)}",
                     Match =
                     {
                         Hosts = SplitHosts(hosts),
                         Path = path,
                         Headers = headerMatches.Count > 0 ? headerMatches.Select(hm => hm.Value).ToArray() : null
                     },
-                    Order = GetLabel(labels, $"{thisRoutePrefix}.Order", DefaultRouteOrder),
+                    Order = order ?? DefaultRouteOrder,
```

Comment:
DefaultRouteOrder is also null, so it looks like we don't need that constant anymore.
```suggestion
                    Order = order,
```